cmd/clusterctl/client/cluster/cert_manager.go

```diff
@@ -401,7 +402,9 @@ func (cm *certManagerClient) getManifestObjs(certManagerConfig config.CertManage
 	// Given that cert manager components yaml are stored in a repository like providers components yaml,
 	// we are using the same machinery to retrieve the file by using a fake provider object using
 	// the cert manager repository url.
-	certManagerFakeProvider := config.NewProvider("cert-manager", certManagerConfig.URL(), "")
+	// We replace "latest" in the url by the wanted version to disable auto version-discovery in repositoryClientFactory.
+	url := strings.Replace(certManagerConfig.URL(), "latest", certManagerConfig.Version(), 1)
+	certManagerFakeProvider := config.NewProvider("cert-manager", url, "")
```

Comment:
This more looks like an ugly workaround. Should we improve the certManagerConfig's api for that?